ui/src/app/shared/components/resource-editor/resource-editor.tsx

```diff
@@ -31,7 +31,7 @@ const LOCAL_STORAGE_KEY = 'ResourceEditorLang';
 export class ResourceEditor<T> extends React.Component<Props<T>, State> {
     private set lang(lang: string) {
         try {
-            this.setState({lang, error: null, value: stringify(parse(this.state.value), lang)});
+            this.setState(state => ({lang, error: null, value: stringify(parse(state.value), lang)}));
```

Comment:
we should not reference `this.state` in `setState(..)` - can create inconsistent state